pkg/test/e2e/nodeport-proxy/services.go

```diff
@@ -105,30 +113,106 @@ func (n *ServiceJig) newNamespaceTemplate() *corev1.Namespace {
 	return ns
 }
 
-func (n *ServiceJig) newServiceTemplate(name, namespace string,
+func (n *ServiceJig) newServiceTemplateNodePort(labels map[string]string) corev1.ServiceSpec {
```

Comment:
I would rather have two specialized methods, say `CreateClusterIPService` and `CreateNodePortService` both calling `newServiceTemplate` and changing the service afterwards.